source/common/filter/tcp_proxy.h

```diff
@@ -145,7 +154,10 @@ class TcpProxy : public Network::ReadFilter,
 
   // Upstream::LoadBalancerContext
   Optional<uint64_t> computeHashKey() override { return {}; }
-  const Router::MetadataMatchCriteria* metadataMatchCriteria() const override { return nullptr; }
+  const Router::MetadataMatchCriteria* metadataMatchCriteria() const override {
+    return config_->metadataMatchCriteria();
```

Comment:
config_ can be nullptr, but not in this case, because WsHandlerImpl overrides this method.  Please add a comment explaining this, or optionally just add a null-check on config_

Reply:
i added a null check because it feels safer (instead of relying on some invariant never changing), but curious how `WsHandlerImpl` comes into play here. is it because `config_` is only null for `WsHandlerImpl`?

Reply:
Very reasonable to just add the null check.

WsHandlerImpl sub-classes TcpProxy, and uses it with a null config_. This is the only case the config is null. I have a TODO to fix that (require non-null config to simplify code).